src/screens/Notifications/AddNotificationEmailScreen.tsx

```diff
@@ -46,8 +46,18 @@ class AddNotificationEmailScreen extends PureComponent<Props, State> {
     email: '',
   };
 
+  forceUpdateComponent = () =>
+    this.props.navigation.addListener('focus', () => {
+      this.setState({ email: '' });
+    });
+
   componentDidMount() {
     this.props.setError('');
+    this.forceUpdateComponent();
+  }
+
+  componentWillUnmount() {
+    this.forceUpdateComponent();
```

Comment:
if you setState on unmount you would see warning "Can't perform a React state update on an unmounted component", beside that there is no point in resetting state after unmount, because this instance won't exist anymore. What you should do on unmount is removing the listener

Reply:
generally unmount listener is good practice. in this case is good also to keep this clear during that flow in the notification is used for 2 scenarios and forces update can break other flows